code/parse/parselo.cpp

```diff
@@ -1665,30 +1666,94 @@ int maybe_convert_foreign_character(int ch)
 }
 
 // Goober5000
-void maybe_convert_foreign_characters(char *line)
+// Yarn - The capacity of out must be at least the value returned by
+// get_converted_string_length(in) (plus one if add_null is true).
+// Returns the number of characters written to out.
+size_t maybe_convert_foreign_characters(char *in, char *out, bool add_null)
```

Comment:
That buffer is passed to `maybe_convert_foreign_characters` as the _input_ buffer. The issue is what's passed as the _output_ buffer, which is (perhaps confusingly) not `outbuf` but rather `mp`, a pointer to some part of `processed_text`, which is also just a pointer.